### ISSUE TITLE
Single launcher

### ISSUE THREAD

**Issue by rivanfebrian123** [sadko4u/lsp-plugins#195]
Hi! Is it possible to make the launchers one? I mean, this is annoying tho
![Screenshot from 2021-05-10 10-08-39](https://user-images.githubusercontent.com/6300910/117601151-c0df0d00-b177-11eb-96f9-20913db77644.png)
Sorry about that

**Comment by sadko4u** [comment]
Hello!

You're not the only who claims about LSP icons flooding GNOME menu entries.
The main problem is that GNOME is not following the XDG standard and not grouping these icons into a single folder.

Since 1.1.30 I decided to delegate the icon installation and maintenance to distribution maintainers by moving icon installation to a dedicated target in the makefile which is not run by default.

Please also see the #193 ticket for the detailed description.

If you have any questions/suggestions, please welcome.

**Comment by rivanfebrian123** [comment]
Hmm, yeah, but I think not providing launchers at all is way too extreme. What about making a single app launcher like "Chrome App Launcher" but it's a window, not a popup. You can imagine it as "GNOME Tweaks" or something

**Comment by riri** [comment]
I agree with @sadko4u , dealing with Gnome is kinda pain in the ass. It is the only environment not providing grouping of desktop files (while there are even special desktop directory annotations for them).

I know it's not the fault of users, they are the *victims*, but I think that it should be more on the Gnome platform to get the request and implement back *directory* views in their shell. That makes a lot of work to make a simple window launcher integrated to the environment, and that's also something that would be necessary for other projects.

P.S.: it's easy for me to say that because I am not a Gnome user lol... but I understand both your need and the project's maintainer issue to implement it.

**Comment by sadko4u** [comment]
By another side, implementing launcher application is a good option but for 1.1.x it is not reasonable.
I think it will be good to implement 1.2.0 (with complete redesign of core modules) release first and then add a launcher application.
So I would consider this issue to be implemented not earlier than 1.2.0 release.

**Comment by rivanfebrian123** [comment]
Oh... I see. Also, I think we should make a guideline about packaging this library. For example:

"You can provide two different packages for LSP, for example: `lsp-plugin-gnome` (or `lsp-plugin-single-launcher`) and `lsp-plugin` (the 'default' one, with multiple launchers)"

**Comment by sadko4u** [comment]
The implementation of single launcher was moved to further releases.

**Comment by sadko4u** [comment]
@bor-T need application concept.

**Comment by sadko4u** [comment]
Single launcher is now available in 1.2.27.

**Comment by dvzrv** [comment]
@sadko4u if I'm not missing something, this launcher application is not installed by default.

On Arch Linux there is now the entire XDG integration but no application.

**Comment by sadko4u** [comment]
Do you specify build targets in `make config FEATURES='...'`?
The new target `launcher` is added there.

**Comment by dvzrv** [comment]
> Do you specify build targets in make config FEATURES='...'?

Yes.

> The new target launcher is added there.

I see. How do I learn about these new targets when they are added?

**Comment by dvzrv** [comment]
Somewhat related: How do I learn what current targets exist?

**Comment by sadko4u** [comment]
You can run `make config` without `ADD_FEATURES` option. After configuring, you'll get list of targets for Linux enabled by default:
```
Host architecture: x86_64/x86_64 (-march=x86-64 -m64)
Architecture:      x86_64/x86_64 (-march=x86-64 -m64)
Features:          clap doc gst jack ladspa launcher lv2 ui vst2 vst3 xdg
Configured OK
```

**Comment by sadko4u** [comment]
Oh, I noticed that forgot to update documentation and `make help` about new feature. Updated just now.
Here's the part output if you call `make help`:
```
Available FEATURES:
  clap                      CLAP plugins
  doc                       Generate standalone HTML documentation
  gst                       GStreamer plugins
  jack                      Standalone JACK plugins
  ladspa                    LADSPA plugins
  launcher                  Build launcher application for standalone JACK plugins
  lv2                       LV2 plugins
  vst2                      VST 2.x plugin binaries
  vst3                      VST 3.x plugin binaries
  xdg                       Desktop integration icons
```
